File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Response/GooglepayEchoResponse.cs
using System.Text;

namespace CsobGatewayClientExample.Communication.DataObjects.Response;

public class GooglepayEchoResponse : SignBase
{
    public int? ResultCode { get; set; }

    public string? ResultMessage { get; set; }

    public GooglepayInitParams? InitParams { get; set; }


    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, Dttm);
        Add(sb, ResultCode);
        Add(sb, ResultMessage);
        if (InitParams != null) Add(sb, InitParams.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/responses/GooglepayEchoResponse.java
package cz.monetplus.mips.eapi.v19.connector.entity.responses;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import cz.monetplus.mips.eapi.v19.connector.entity.SignBase;
import lombok.*;

@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor @NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class GooglepayEchoResponse extends SignBase {

    private Integer resultCode;
    private String resultMessage;
    private GooglepayInitParams initParams;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, getDttm());
        add(sb, getResultCode());
        add(sb, getResultMessage());
        if (getInitParams() != null) {
            add(sb, getInitParams().toSign());
        }
        deleteLast(sb);
        return sb.toString();
    }
}
